File: Data Structures & Algorithms/duplicate-integer/submission-3.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        
    Set<Integer> numbers = new HashSet<>(); 

    for (int num : nums) {
        numbers.add(num); 
    }

    if (nums.length > numbers.size()) {return true; } 

    return false; 

    }  

}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # we can subtract current val from the overall targer integer
        # then we can try and see if the value exists in the hashmap
        # if exists, we can return the indices of the two values
        
        hashmap = {}
        
        # store value → index
        for i in range(len(nums)):
                    value = nums[i]
                    hashmap[value] = i

        # Step 2: Loop again to find the complement
        for i in range(len(nums)):
            value = nums[i]
            search_val = target - value
            if search_val in hashmap and hashmap[search_val] != i:
                return [i, hashmap[search_val]]
